tests/integration/buildcmd/test_build_cmd.py

```diff
@@ -528,6 +528,97 @@ def _verify_built_artifact(self, build_dir, function_logical_id, expected_files)
         self.assertEqual(actual_files, expected_files)
 
 
+@skipIf(
+    ((IS_WINDOWS and RUNNING_ON_CI) and not CI_OVERRIDE),
+    "Skip build tests on windows when running in CI unless overridden",
+)
+class TestBuildCommand_Go_Modules(BuildIntegBase):
+
+    FUNCTION_LOGICAL_ID = "Function"
+    EXPECTED_FILES_PROJECT_MANIFEST = {"hello-world"}
+
+    @pytest.mark.flaky(reruns=3)
+    @parameterized.expand([("go1.x", "Go", None), ("go1.x", "Go", "debug")])
+    def test_with_go(self, runtime, code_uri, mode):
+        overrides = {"Runtime": runtime, "CodeUri": code_uri, "Handler": "hello-world"}
+        cmdlist = self.get_command_list(use_container=False, parameter_overrides=overrides)
+
+        # Need to pass GOPATH ENV variable to match the test directory when running build
+
+        LOG.info("Running Command: {}".format(cmdlist))
+        LOG.info("Running with SAM_BUILD_MODE={}".format(mode))
+
+        newenv = os.environ.copy()
+        if mode:
+            newenv["SAM_BUILD_MODE"] = mode
```

Comment:
This will produce a binary but the binary is not debugable. Not sure if we want to add this test now or when we enable debug builds for go. Just wanted to call it out.